rasterio/_io.pyx

```diff
@@ -1122,6 +1122,56 @@ cdef class RasterReader(_base.DatasetReader):
         return out
 
 
+    def dataset_mask(self, *args, window=None, boundless=False):
+        """Calculate the dataset's 2D mask. Derived from the individual band masks
+        provided by read_masks().
+
+        Parameters
+        ----------
+        position args ignored
+        window and boundless are passed directly to read_masks()
+        
+        Returns
+        -------
+        ndarray, shape=(self.height, self.width), dtype='uint8'
+        0 = nodata, 255 = valid data
+
+        The dataset mask is calculate based on the individual band masks according to
+        the following logic, in order of precedence:
+
+        1. If a .msk file, dataset-wide alpha or internal mask exists,
+           it will be used as the dataset mask.
+        2. If an 4-band RGBA with a shadow nodata value,
+           band 4 will be used as the dataset mask.
+        3. If a nodata value exists, use the binary OR (|) of the band masks
+        4. If no nodata value exists, return a mask filled with 255
+
+        Note that this differs from read_masks and GDAL RFC15
+        in that it applies per-dataset, not per-band
+        (see https://trac.osgeo.org/gdal/wiki/rfc15_nodatabitmask)
+        """
+        del args
```

Comment:
👍  I was trying to get fancy by requiring that parameter be kwargs. Not necessary so I'll remove it.